stdlib/TOML/src/print.jl

```diff
@@ -104,6 +104,15 @@ function printvalue(f::MbyFunc, io::IO, value::TOMLValue)
     error("internal error in TOML printing, unhandled value")
 end
 
+function print_integer(io::IO, value::Integer)
+    value isa Signed && return Base.show(io, value)
+    # unsigned integers are printed as hex
+    n = Base.top_set_bit(value) ÷ 4 + 1
+    isodd(n) && (n += 1)
```

Comment:
why do odd numbers need an extra digit?

Reply:
https://github.com/JuliaLang/julia/pull/47903#discussion_r1063606076

Reply:
This could also be done by using `ndigits(value, base=256)` and then doubling the number of digits below.